### DISCUSSION TITLE
Streamlit Booth Slider Ideas

### DISCUSSION THREAD

**Discussion by kjaymiller** [BlackPythonDevs/blackpythondevs#36] [Ideas]
Repo: https://github.com/BlackPythonDevs/streamlit-booth-slider

This is a discussion about what can be included in the project.

**Comment by kjaymiller** [comment]
## Tabs and sections

There are so many things that we'd like to share here are a few ideas that I think we can include with good navigation.

I believe that if you create a sliding layout that vertically will show different pages in a single section and horizontally across different sections we can have an auto-sliding presentation that is easy to navigate.

**Reply by ronnzw** [reply]
It can be helpful to add visualisation screenshots if you have one. It doesn't have to be something specific but just a guideline.

**Reply by kjaymiller** [reply]
Honestly I'm not familiar with streamlit enough to know if it's possible but I have been playing with the idea of multiple pages in streamlit so I can link to a particular technology - https://docs.streamlit.io/get-started/tutorials/create-a-multipage-app

**Reply by ronnzw** [reply]
i have been playing around with the tabs and l think they can works but we need to answer a few questions:

1. What will each tab represent? will it be a project or an individual's profile or future plans?
2. Should folks be able to scroll vertically or should we make it a view height of 100% with no option to scroll
3. I haven't come across a slider component meaning we will need to build our own, maybe through react or just pure JS, will it not limit other folks from contributing?

<img width="1458" alt="Screenshot 2024-03-07 at 18 13 25" src="https://github.com/BlackPythonDevs/.maintainers/assets/17087071/1fb813fb-69f0-40c0-95f8-f7c043bd9f6a">

**Reply by kjaymiller** [reply]
**Home** - Black Python Devs at PyCon and how to support
**About** - Copied from our about page
**Projects & Initiatives** can be a list of things with a learn more  
**I'm hoping we can do a Swag Page as well to purchase things like shirts and stuff** (Working on this)

**Comment by kjaymiller** [comment]
## Auto-progression and pause

The idea is to make sure that folks can navigate through the application or they can let it auto-scroll on their own.

My idea is to add keybindings two change between [tabs and sections](https://github.com/BlackPythonDevs/.maintainers/discussions/36#discussioncomment-8657716). There should be a call to action on each section.

We can use QR codes to help make it easy to follow folks and learn more about a project. That means we need to pause the page so folks can take their time to follow things.

### Keybindings

- H, L (or A,D or Left Arrow, Right Arrow) move across tabs
- J, K (W, S or Up Arrow, Down Arrow) navigate content sections
- space - play/pause

**Reply by ronnzw** [reply]
Tabs come with configured arrow keys already so we should be good.
